Index: .classpath
===================================================================
--- .classpath	(revision 1115)
+++ .classpath	(working copy)
@@ -7,5 +7,6 @@
 	<classpathentry kind="lib" path="dist/libs/mysql-connector-java-5.1.38-bin.jar"/>
 	<classpathentry including="**/*.java" kind="src" path="dist/game/data/scripts"/>
 	<classpathentry kind="src" path="java"/>
+	<classpathentry kind="lib" path="dist/libs/L2J_EventEngine.jar"/>
 	<classpathentry kind="output" path="bin"/>
 </classpath>
Index: java/com/l2jmobius/gameserver/model/olympiad/AbstractOlympiadGame.java
===================================================================
--- java/com/l2jmobius/gameserver/model/olympiad/AbstractOlympiadGame.java	(revision 1115)
+++ java/com/l2jmobius/gameserver/model/olympiad/AbstractOlympiadGame.java	(working copy)
@@ -42,6 +42,8 @@
 import com.l2jmobius.gameserver.network.serverpackets.SkillCoolTime;
 import com.l2jmobius.gameserver.network.serverpackets.SystemMessage;
 
+import net.sf.eventengine.EventEngineManager;
+
 /**
  * @author godson, GodKratos, Pere, DS
  */
@@ -157,6 +159,12 @@
 			return SystemMessage.getSystemMessage(SystemMessageId.YOUR_OPPONENT_DOES_NOT_MEET_THE_REQUIREMENTS_TO_DO_BATTLE);
 		}
 		
+		// L2J EventEngine
+		if (EventEngineManager.getInstance().isPlayableInEvent(player))
+		{
+			return SystemMessage.getSystemMessage(SystemMessageId.YOUR_OPPONENT_DOES_NOT_MEET_THE_REQUIREMENTS_TO_DO_BATTLE);
+		}
+		
 		return null;
 	}
 	
Index: java/com/l2jmobius/gameserver/network/clientpackets/UseItem.java
===================================================================
--- java/com/l2jmobius/gameserver/network/clientpackets/UseItem.java	(revision 1115)
+++ java/com/l2jmobius/gameserver/network/clientpackets/UseItem.java	(working copy)
@@ -46,6 +46,8 @@
 import com.l2jmobius.gameserver.network.serverpackets.ItemList;
 import com.l2jmobius.gameserver.network.serverpackets.SystemMessage;
 
+import net.sf.eventengine.EventEngineManager;
+
 public final class UseItem extends L2GameClientPacket
 {
 	private static final String _C__19_USEITEM = "[C] 19 UseItem";
@@ -170,6 +172,12 @@
 			}
 		}
 		
+		// L2J EventEngine
+		if (EventEngineManager.getInstance().listenerOnUseItem(activeChar, item.getItem()))
+		{
+			return;
+		}
+		
 		// If the item has reuse time and it has not passed.
 		// Message from reuse delay must come from item.
 		final int reuseDelay = item.getReuseDelay();
